bind.go

```diff
@@ -39,10 +41,26 @@ type maybeLineage struct {
 
 	allv []SyntacticVersion
 
+	implens []ImperativeLens
+
+	// translation plan, tracks whether each lens is handled in cue or go
+	transplan map[lensID]bool
+
 	// The raw input value is the root of a package instance
 	// rawIsPackage bool
 }
 
+// to, from
+type lensID [2]SyntacticVersion
+
+func lid(to, from SyntacticVersion) lensID {
```

Comment:
similarly, reason for not defining this as from, to? This seems backward, I would say "I am going from `a` to `b`", not "I am going to `b` from `a`".

Reply:
Not a good one, no. it's all a weird artifact of probably mistakenly trying to be consistent with [the expected sort ordering in lenses](https://github.com/grafana/thema/blob/2a57611eed5ce3fcb08eef5c68cb4c05de2b131f/lineage.cue#L49-L75), where we sort first by `to` and then by `from`. (That declared sort ordering requirement only applies for CUE lenses, not for these Go things)

Reply:
Just for curiosity; Could you give more context about the reasons behind such requirement, please? Thanks!

Reply:
Requiring the inputs to be manually sorted by the user was the main way we got away from the need to dynamically sort them in CUE - which was the main source of our performance issues fixed in #173

Reply:
Sorry, I mean specifically about the order: _first by `to` and then by `from`_

Reply:
ahhh - i'm not attached to that as an ordering, either. My motivation was just that, assuming an ascending version sort order (same as with `schemas`), it seemed preferable to have forward lenses precede reverse ones in the list.